src/modules/Elsa.Workflows.Core/Contracts/IActivityRegistryLookupService.cs

```diff
@@ -0,0 +1,38 @@
+using Elsa.Workflows.Models;
+
+namespace Elsa.Workflows.Contracts;
+
+/// <summary>
+/// Represents a service for looking up activity descriptors.
+/// </summary>
+public interface IActivityRegistryLookupService
+{
+    /// <summary>
+    /// Returns the activity descriptor for the specified activity type.
+    /// </summary>
+    /// <param name="type">The activity type.</param>
+    /// <returns>The activity descriptor for the specified activity type.</returns>
+    Task<ActivityDescriptor?> Find(string type);
```

Comment:
Please mind the naming convention to apply the `Async` suffix when declaring async methods.